shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BlockChain.hs

```diff
@@ -593,27 +682,29 @@ blockDecoder lax = annotatorSlice $
     txns <- txSeqDecoder lax
     pure $ Block' <$> header <*> txns
 
+-- | Decode a TxSeq, used in decoding a Block.
 txSeqDecoder ::
-  ( ToCBOR (Core.TxBody era),
-    ToCBOR (Core.AuxiliaryData era),
-    ToCBOR (Core.Script era),
-    BlockAnn era,
-    ValidateScript era
-  ) =>
+  forall era.
+  (Era era, BlockAnn era) =>
   Bool ->
   forall s. Decoder s (Annotator (TxSeq era))
 txSeqDecoder lax = do
   (bodies, bodiesAnn) <- withSlice $ decodeSeq fromCBOR
-  (wits, witsAnn) <- withSlice $ decodeSeq decodeWits
+  (wits, witsAnn) <- withSlice $ decodeSeq fromCBOR
   let b = length bodies
+      inRange x = (0 <= x) && (x <= (b -1))
       w = length wits
-
-  (metadata, metadataAnn) <-
-    withSlice $
-      constructMetadata b
-        <$> decodeMap fromCBOR fromCBOR
-  let m = length metadata
-
+  (metadata, metadataAnn) <- withSlice $
+    do
+      m <- decodeMap fromCBOR fromCBOR
+      unless
+        (lax || all inRange (Map.keysSet m))
```

Comment:
note: does this reject any existing blocks?

Reply:
Left a note to test for this.